jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/AbstractHTTP2ServerConnectionFactory.java

```diff
@@ -245,18 +252,43 @@ protected ServerParser newServerParser(Connector connector, ServerParser.Listene
         return new ServerParser(connector.getByteBufferPool(), listener, getMaxDynamicTableSize(), getHttpConfiguration().getRequestHeaderSize());
     }
 
-    private class ConnectionListener implements Connection.Listener
+    private class HTTP2SessionContainer implements Connection.Listener, Dumpable
```

Comment:
make this a `ManagedObject` and `dump()` a `ManagedOperation`, so we can dump just the sessions from JMX.  Also make the size of the sessions collection a managed attribute.

Reply:
Good suggestions, will do.